plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityExploredLocations.java

```diff
@@ -0,0 +1,48 @@
+package com.denizenscript.denizen.objects.properties.entity;
+
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import org.bukkit.Location;
+import org.bukkit.entity.Sniffer;
+
+public class EntityExploredLocations extends EntityProperty<ListTag> {
+
+    // <--[property]
+    // @object EntityTag
+    // @name explored_locations
+    // @input ListTag(LocationTag)
+    // @description
+    // If the entity is a sniffer, controls the locations that it has explored. Note that if the sniffer is not in the same world as the inputted LocationTag(s), then the LocationTag(s) will not be added to the list of explored locations.
+    // -->
+
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof Sniffer;
+    }
+
+    @Override
+    public ListTag getPropertyValue() {
+        ListTag locations = new ListTag();
+        for (Location location : as(Sniffer.class).getExploredLocations()) {

```

Comment:
I believe this can be compressed to a single clean line with the `new ListTag(collection, func)` constructor